src/main/java/no/fintlabs/role/RoleService.java

```diff
@@ -41,93 +45,107 @@ public Role save(Role role) {
         }
     }
 
-    public Flux<Role> getAllRoles() {
-        List<Role> allRoles = roleRepository.findAll().stream().collect(Collectors.toList());
-        return Flux.fromIterable(allRoles);
+    public List<Role> getAllRoles() {
+        return roleRepository.findAll().stream().collect(Collectors.toList());
     }
 
     public Mono<Role> createNewRole(Role role) {
         Role newRole = roleRepository.save(role);
         return Mono.just(newRole);
     }
 
-    public Mono<Role> findRoleById(Long id) {
-        Role role = roleRepository.findById(id).orElse(new Role());
-        return Mono.just(role);
+    public Role findRoleById(Long id) {
+        return roleRepository.findById(id).orElse(new Role());
     }
 
-    public Mono<Role> findRoleByRoleId(String roleId) {
-        Role role = roleRepository.findByRoleId(roleId).orElse(new Role());
-        return Mono.just(role);
+    public Role findRoleByRoleId(String roleId) {
+        return roleRepository.findByRoleId(roleId).orElse(new Role());
     }
 
-    public Mono<Role> findRoleByResourceId(String id) {
-        Role role = roleRepository.findByResourceId(id).orElse(new Role());
-        return Mono.just(role);
+    public Role findRoleByResourceId(String id) {
+        return roleRepository.findByResourceId(id).orElse(new Role());
+
     }
 
-    public Mono<DetailedRole> GetDetailedRoleById(Long id) {
-        return Mono.just(roleRepository.findById(id)
+    public DetailedRole GetDetailedRoleById(Long id) {
+        return roleRepository.findById(id)
                 .map(Role::toDetailedRole)
-                .orElse(new DetailedRole())
+                .orElse(new DetailedRole()
         );
     }
 
```

Comment:
getSimpleRoles er litt lang og litt mye if/else. Tenker den fint kan deles opp med et par private metoder. For eksempel:

```suggestion
public List<SimpleRole> getSimpleRoles(
            FintJwtEndUserPrincipal principal,
            String search,
            List<String> orgUnits,
            List<String> orgUnitsInScope,
            String roleType,
            Boolean aggRoles
    ) {
        List<String> orgUnitsInSearch = getOrgUnitsInSearch(orgUnits, orgUnitsInScope);
        List<Role> roles = getRoles(search, orgUnitsInSearch, roleType, aggRoles);
        return roles.stream().map(Role::toSimpleRole).collect(Collectors.toList());
    }

    private List<String> getOrgUnitsInSearch(List<String> orgUnits, List<String> orgUnitsInScope) {
        if (orgUnits == null) {
            log.info("OrgUnits parameter is empty, using orgunits from scope {} in search", orgUnitsInScope);
            return orgUnitsInScope;
        } else {
            log.info("OrgUnits parameter list: {}", orgUnits);
            List<String> filteredOrgUnits = orgUnits.stream()
                    .filter(orgUnitsInScope::contains)
                    .collect(Collectors.toList());
            log.info("OrgUnits in search: {}", filteredOrgUnits);
            return filteredOrgUnits;
        }
    }

    private List<Role> getRoles(String search, List<String> orgUnitsInSearch, String roleType, Boolean aggRoles) {
        if ("ALLTYPES".equals(roleType)) {
            if (aggRoles == null) {
                return roleRepository.getRolesByNameOrgunitsAggregated(search, orgUnitsInSearch);
            }
            return roleRepository.getRolesByNameOrgunitsAggregated(search, orgUnitsInSearch, aggRoles);
        }

        if (aggRoles == null) {
            return roleRepository.getRolesByNameTypeOrgunitsAggregated(search, roleType, orgUnitsInSearch);
        }
        return roleRepository.getRolesByNameTypeOrgunitsAggregated(search, roleType, orgUnitsInSearch, aggRoles);
    }
```